src/cpptools.ts

```diff
@@ -132,13 +132,23 @@ export class CppConfigurationProvider implements cpt.CustomConfigurationProvider
    */
   private _lastUpdateSucceeded = true;
 
+  private _workspaceBrowseConfiguration:cpt.WorkspaceBrowseConfiguration = {browsePath:[]};
+
+  private _cachedSourceFileConfiguration: cpt.SourceFileConfigurationItem|undefined;
```

Comment:
You can use a shorthand for `T | undefined`:

```suggestion
  private _cachedSourceFileConfiguration?: cpt.SourceFileConfigurationItem;
```

Reply:
Nice idea